libc/startup/linux/CMakeLists.txt

```diff
@@ -1,3 +1,42 @@
+# This function merges multiple objects into a single relocatable object
+#                     cc -r obj1.o obj2.o -o obj.o
+# A relocatable object is an object file that is not fully linked into an
+# executable or a shared library. It is an intermediate file format that can
+# be passed into the linker.
+# A crt object have arch-specific code and arch-agnostic code. To reduce code
+# cohesion, the implementation is splitted into multiple units. As a result,
+# we need to merge them into a single relocatable object.
+function(merge_relocatable_object name)
+  set(obj_list "")
+  set(fq_link_libraries "")
+  get_fq_deps_list(fq_dep_list ${ARGN})
+  foreach(target IN LISTS fq_dep_list)
+    list(APPEND obj_list "$<TARGET_OBJECTS:${target}>")
+    get_target_property(libs ${target} DEPS)
+    list(APPEND fq_link_libraries "${libs}")
+  endforeach()
+  get_fq_target_name(${name} fq_name)
+  add_custom_command(
```

Comment:
Were you unable to do this by importing an object library or similar? I feel like we should just be able to make this an executable that imports from an object library.

Reply:
Just to confirm, do you mean treat `crt1.o` as an executable rather than an object library?

Reply:
Just my thoughts, that if we create an executable but pass `-Wl,-r` to the linker it'll output as an object. Then we can jsut set the output name property.

Reply:
There are internal uses of crt1 as an object library. See integration tests.

Reply:
Maybe I'm missing context. Why do we need to use a merged version of the startup code for testing? I thought this was just for exporting.

Reply:
That may just work, but won't it be nice to test exactly the same object as what the installation will ship?

Reply:
We could also link it in by filename, though it's a little uglier. In general I'm just averse to `add_custom_command` things that have output, they're usually a pain to deal with.

We also need to check for support from the given linker. Usually that involves sending a test program to some CMake magic function that checks if compiling it with the given flags failed.

Reply:
I will add the check and look for a better solution than custom command.